sim/battle.ts

```diff
@@ -1022,15 +1022,12 @@ export class Battle extends Dex.ModdedDex {
 		}
 		}
 
-		let allRequestsMade = true;
-		for (const request of requests) {
-			if (!request) allRequestsMade = false;
-		}
+		let allRequestsMade = !requests.some(request => !request);
```

Comment:
```

```suggestion
		let allRequestsMade = requests.every(request => request);
```